pkg/provider/cloud/eks/provider.go

```diff
@@ -432,66 +417,61 @@ func ListUpgrades(ctx context.Context,
 	switch status {
-	case string(resources.CreatingEKSState):
+	case ekstypes.ClusterStatusCreating:
 		return apiv2.ProvisioningExternalClusterState
-	case string(resources.PendingEKSState):
+	case ekstypes.ClusterStatusPending:
 		return apiv2.ProvisioningExternalClusterState
-	case string(resources.ActiveEKSState):
+	case ekstypes.ClusterStatusActive:
 		return apiv2.RunningExternalClusterState
-	case string(resources.UpdatingEKSState):
+	case ekstypes.ClusterStatusUpdating:
 		return apiv2.ReconcilingExternalClusterState
-	case string(resources.DeletingEKSState):
+	case ekstypes.ClusterStatusDeleting:
 		return apiv2.DeletingExternalClusterState
-	case string(resources.FailedEKSState):
+	case ekstypes.ClusterStatusFailed:
 		return apiv2.ErrorExternalClusterState
 	default:
 		return apiv2.UnknownExternalClusterState
 	}
 }
 
-func ConvertMDStatus(status string) apiv2.ExternalClusterMDState {
+func ConvertMDStatus(status ekstypes.NodegroupStatus) apiv2.ExternalClusterMDState {
 	switch status {
-	case string(resources.CreatingEKSMDState):
+	case ekstypes.NodegroupStatusCreating:
 		return apiv2.ProvisioningExternalClusterMDState
-	case string(resources.ActiveEKSMDState):
+	case ekstypes.NodegroupStatusActive:
 		return apiv2.RunningExternalClusterMDState
-	case string(resources.UpdatingEKSMDState):
+	case ekstypes.NodegroupStatusUpdating:
 		return apiv2.ReconcilingExternalClusterMDState
-	case string(resources.DeletingEKSMDState):
+	case ekstypes.NodegroupStatusDeleting:
 		return apiv2.DeletingExternalClusterMDState
-	case string(resources.CreateFailedEKSMDState):
+	case ekstypes.NodegroupStatusCreateFailed:
 		return apiv2.ErrorExternalClusterMDState
-	case string(resources.DeleteFailedEKSMDState):
+	case ekstypes.NodegroupStatusDeleteFailed:
 		return apiv2.ErrorExternalClusterMDState
-	case string(resources.DegradedEKSMDState):
+	case ekstypes.NodegroupStatusDegraded:
 		return apiv2.ErrorExternalClusterMDState
 	default:
 		return apiv2.UnknownExternalClusterMDState
 	}
 }
 
 func ValidateCredentials(ctx context.Context, credential resources.EKSCredential) error {
-	client, err := awsprovider.GetClientSet(credential.AccessKeyID, credential.SecretAccessKey, "", "", credential.Region)
+	client, err := awsprovider.GetClientSet(ctx, credential.AccessKeyID, credential.SecretAccessKey, "", "", credential.Region)
 	if err != nil {
 		return err
 	}
-	_, err = ListClusters(client)
+	_, err = ListClusters(ctx, client)
 
 	return DecodeError(err)
 }
 
 func DecodeError(err error) error {
 	// Generic AWS Error with Code, Message, and original error (if any).
-	var aerr awserr.Error
+	var aerr smithy.APIError
 	if errors.As(err, &aerr) {
-		// A service error occurred.
-		var reqErr awserr.RequestFailure
-		if errors.As(aerr, &reqErr) {
-			return utilerrors.New(reqErr.StatusCode(), reqErr.Message())
-		}
-		return errors.New(aerr.Message())
+		return utilerrors.New(500, fmt.Sprintf("%s: %s", aerr.ErrorCode(), aerr.ErrorMessage()))
```

Comment:
why hardcoded status code to 500?

Reply:
Good catch! I think we should get rid of this error handling code alltogether, to quote my commit:

> smithy.APIError implementations all seem to be doing the same, if not better string marshalling that we would do.
>
> Also using HTTP 500 (internal server error) for every possible fault is not ideal (ValidateCredentials() should not error out with an internal server error on bad credentials), and in general the code in the provider package should not return HTTP-related things, as controllers also use it.
>
> I removed the DecodeError() entirely, as it seems to serve no purpose anymore with SDK v2. :)

Reply:
but, this part of code from external cluster controller expects error to be an `utilerrors.HTTPError` :
https://github.com/kubermatic/kubermatic/blob/master/pkg/controller/master-controller-manager/external-cluster/cluster_controller.go#L106

either we can modify controller here, or I can create a separate pr as this will not immediately affect the controller unless `notFound` clusters

Reply:
@imharshita can you please take another look, Christoph fixed this in https://github.com/kubermatic/kubermatic/pull/10922/commits/628c8727b937b7eb338b1eb009283f3ca61d56da I think.

Reply:
Sorry, seems I misunderstood this conversation. Looking.

Reply:
@imharshita I reworked the DecodeError function, it should now return what is expected by the controller code, if that information was included in the returned error. Please take another look.

Reply:
/lgtm 
/approve
@embik  regarding hardcoding 500 code in DecodeError , I will fix that in a separate pr.

Reply:
the hardcoded 500 error is already resolved. We only fall back to 500 if no HTTP information was included in the error.